src/Numerics/DGmethods/remainder.jl

```diff
@@ -0,0 +1,610 @@
+
+vars_state_auxiliary(rem_balance_law::RemBL, FT) =
+    vars_state_auxiliary(rem_balance_law.main, FT)
+
+vars_integrals(rem_balance_law::RemBL, FT) =
+    vars_integrals(rem_balance_law.main, FT)
+
+vars_reverse_integrals(rem_balance_law::RemBL, FT) =
+    vars_integrals(rem_balance_law.main, FT)
+
+vars_gradient_laplacian(rem_balance_law::RemBL, FT) =
+    vars_gradient_laplacian(rem_balance_law.main, FT)
+
+vars_hyperdiffusive(rem_balance_law::RemBL, FT) =
+    vars_hyperdiffusive(rem_balance_law.main, FT)
+
+update_auxiliary_state!(dg::DGModel, rem_balance_law::RemBL, args...) =
+    update_auxiliary_state!(dg, rem_balance_law.main, args...)
+
+update_auxiliary_state_gradient!(dg::DGModel, rem_balance_law::RemBL, args...) =
+    update_auxiliary_state_gradient!(dg, rem_balance_law.main, args...)
+
+integral_load_auxiliary_state!(rem_balance_law::RemBL, args...) =
+    integral_load_auxiliary_state!(rem_balance_law.main, args...)
+
+integral_set_auxiliary_state!(rem_balance_law::RemBL, args...) =
+    integral_set_auxiliary_state!(rem_balance_law.main, args...)
+
+reverse_integral_load_auxiliary_state!(rem_balance_law::RemBL, args...) =
+    reverse_integral_load_auxiliary_state!(rem_balance_law.main, args...)
+
+reverse_integral_set_auxiliary_state!(rem_balance_law::RemBL, args...) =
+    reverse_integral_set_auxiliary_state!(rem_balance_law.main, args...)
+
+transform_post_gradient_laplacian!(rem_balance_law::RemBL, args...) =
+    transform_post_gradient_laplacian!(rem_balance_law.main, args...)
+
+flux_second_order!(rem_balance_law::RemBL, args...) =
+    flux_second_order!(rem_balance_law.main, args...)
+
+compute_gradient_argument!(rem_balance_law::RemBL, args...) =
+    compute_gradient_argument!(rem_balance_law.main, args...)
+
+compute_gradient_flux!(rem_balance_law::RemBL, args...) =
+    compute_gradient_flux!(rem_balance_law.main, args...)
+
+boundary_state!(nf, rem_balance_law::RemBL, args...) =
+    boundary_state!(nf, rem_balance_law.main, args...)
+
+init_state_auxiliary!(rem_balance_law::RemBL, args...) =
+    init_state_auxiliary!(rem_balance_law.main, args...)
+
+init_state_conservative!(rem_balance_law::RemBL, args...) =
+    init_state_conservative!(rem_balance_law.main, args...)
+
+"""
+    function flux_first_order!(
+        rem_balance_law::RemBL,
+        flux::Grad,
+        state::Vars,
+        aux::Vars,
+        t::Real,
+        directions,
+    )
+
+Evaluate the remainder flux by first evaluating the main flux and subtracting
+the subcomponent fluxes.
+
+
+Only models which have directions that are included in the `directions` tuple
+are evaluated. When these models are evaluated the models underlying `direction`
+is passed (not the original `directions` argument).
+"""
+function flux_first_order!(
+    rem_balance_law::RemBL,
+    flux::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+    ::Dirs,
+) where {NumDirs, Dirs <: NTuple{NumDirs, Direction}}
+    m = getfield(flux, :array)
```

Comment:
Is this any different from using `parent` in this context?

Reply:
probably not. I was just copying what was already in the file.

Reply:
Fixed this. Thanks for catching it.